6.bulk_Morphology_Elastic_Network/3.model_coefficients/scripts/3.model_parameter_visualization.r

```diff
@@ -0,0 +1,42 @@
+suppressPackageStartupMessages(suppressWarnings(library(ggplot2)))
+
+cell_type = "PBMC"
+# path set
+input_file_path <- file.path(paste0("../results/","regression/",cell_type))
+# read in the data
+output_path <- file.path(paste0("../figures/","regression/",cell_type,"/"))
+# create output directory if it doesn't exist
+dir.create(output_path, recursive = TRUE, showWarnings = FALSE)
+
+files <- list.files(path = input_file_path, pattern = "*.csv", full.names = TRUE)
+
+# define empty df with column names
+model_df <- data.frame(
```

Comment:
Found this really neat to see, love the declarative nature here. Completely optional: is there a way to ensure the column types are received as described (casting or raising an error if not)? What would happen if the data value types were out of alignment for some reason?  [Pandera](https://github.com/unionai-oss/pandera) comes to mind within the Python landscape, but I wasn't sure if there was an R equivalent.

Reply:
@d33bs  I believe R raises an error something along the lines of  "cannot coerce data type" if a column is predefined as a `numeric` type and a `string` is passed.  The predefined column types help with ensuring that the `numeric` values stay numeric. There is a level deeper here in R too with `dbl` floats and `int` and such.

Reply:
Thanks for the reply on this - interesting stuff!